libs/core/src/lib/select/select-proxy.service.ts

```diff
@@ -0,0 +1,9 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { OptionStatusChange } from './select.component';
+
+@Injectable()
+export class SelectProxy {
```

Comment:
What is the added value moving all this to extra service ?

Reply:
Previous implementation included cyclic dependency:
- Select to Option with `ContentChildren(OptionComponent)`
- Option to Select with element injector `constructor(SelectComponent)`

My first try was implementing a communication between Select an Option based on:
- Select to Option with `ContentChildren(OptionComponent)`
- Select subscribes to `Option.onOptionSelect()` emitter
but once again I couldn't escape change detection error and the code have got more complicated because of access to Select->Option in late change detection phase.

I decided to create communication proxy between Select and Option to escape cyclic dependency.